nvflare/client/api.py

```diff
@@ -13,61 +13,55 @@
 # limitations under the License.
 
 import logging
-import os
-from enum import Enum
 from typing import Any, Dict, Optional
 
 from nvflare.apis.analytix import AnalyticsDataType
 from nvflare.app_common.abstract.fl_model import FLModel
-from nvflare.fuel.data_event.data_bus import DataBus
 
-from .api_spec import CLIENT_API_KEY, CLIENT_API_TYPE_KEY, APISpec
-from .ex_process.api import ExProcessClientAPI
+from .api_context import APIContext
 
+context_dict = {}
+default_context = None
 
-class ClientAPIType(Enum):
-    IN_PROCESS_API = "IN_PROCESS_API"
-    EX_PROCESS_API = "EX_PROCESS_API"
 
-
-client_api: Optional[APISpec] = None
-data_bus = DataBus()
-
-
-def init(rank: Optional[str] = None):
+def init(rank: Optional[str] = None, config_file: Optional[str] = None) -> APIContext:
     """Initializes NVFlare Client API environment.
 
     Args:
         rank (str): local rank of the process.
             It is only useful when the training script has multiple worker processes. (for example multi GPU)
+        config_file (str): client api configuration.
 
     Returns:
-        None
+        APIContext
     """
-    api_type_name = os.environ.get(CLIENT_API_TYPE_KEY, ClientAPIType.IN_PROCESS_API.value)
-    api_type = ClientAPIType(api_type_name)
-    global client_api
-    if client_api is None:
-        if api_type == ClientAPIType.IN_PROCESS_API:
-            client_api = data_bus.get_data(CLIENT_API_KEY)
-        else:
-            client_api = ExProcessClientAPI()
-        client_api.init(rank=rank)
+    global context_dict
```

Comment:
The whole thing should be placed within a lock in case the app calls flare.init() at the same time from different threads.

Reply:
fixed as suggested.